### ISSUE TITLE
AttributeError: 'str' object has no attribute 'absolute'

### ISSUE THREAD

**Issue by a4dyn** [crytic/slither#900]
When I try to analyze pancakeswap masterchef with slither, the following error occurs:

```
Compilation warnings/errors on crytic-export/etherscan-contracts/0x73feaa1ee314f8c655e354234017be2193c9e24e.bscscan.com-MasterChef.sol:
Warning: SPDX license identifier not provided in source file. Before publishing, consider adding a comment containing "SPDX-License-Identifier: <SPDX-License>" to each source file. Use "SPDX-License-Identifier: UNLICENSED" for non-open-source code. Please see https://spdx.org for more information.
--> crytic-export/etherscan-contracts/0x73feaa1ee314f8c655e354234017be2193c9e24e.bscscan.com-MasterChef.sol

Warning: Documentation tag on non-public state variables will be disallowed in 0.7.0. You will need to use the @dev tag explicitly.
   --> crytic-export/etherscan-contracts/0x73feaa1ee314f8c655e354234017be2193c9e24e.bscscan.com-MasterChef.sol:964:5:
    |
964 |     /// @notice A record of each accounts delegate
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Warning: Documentation tag on non-public state variables will be disallowed in 0.7.0. You will need to use the @dev tag explicitly.
    --> crytic-export/etherscan-contracts/0x73feaa1ee314f8c655e354234017be2193c9e24e.bscscan.com-MasterChef.sol:1228:5:
     |
1228 |     /// @notice A record of each accounts delegate
     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^


Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/slither/__main__.py", line 723, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
  File "/usr/local/lib/python3.9/site-packages/slither/__main__.py", line 82, in process_all
    ) = process_single(compilation, args, detector_classes, printer_classes)
  File "/usr/local/lib/python3.9/site-packages/slither/__main__.py", line 65, in process_single
    slither = Slither(target, ast_format=ast, **vars(args))
  File "/usr/local/lib/python3.9/site-packages/slither/slither.py", line 81, in __init__
    parser.parse_top_level_from_loaded_json(ast, path)
  File "/usr/local/lib/python3.9/site-packages/slither/solc_parsing/slither_compilation_unit_solc.py", line 180, in parse_top_level_from_loaded_json
    pragma.set_offset(top_level_data["src"], self._compilation_unit)
  File "/usr/local/lib/python3.9/site-packages/slither/core/source_mapping/source_mapping.py", line 124, in set_offset
    self._source_mapping = self._convert_source_mapping(offset, compilation_unit)
  File "/usr/local/lib/python3.9/site-packages/slither/core/source_mapping/source_mapping.py", line 101, in _convert_source_mapping
    (lines, starting_column, ending_column) = self._compute_line(
  File "/usr/local/lib/python3.9/site-packages/slither/core/source_mapping/source_mapping.py", line 40, in _compute_line
    start_line, starting_column = compilation_unit.core.crytic_compile.get_line_from_offset(
  File "/usr/local/lib/python3.9/site-packages/crytic_compile-0.2.0-py3.9.egg/crytic_compile/crytic_compile.py", line 235, in get_line_from_offset
    self._get_cached_offset_to_line(file)
  File "/usr/local/lib/python3.9/site-packages/crytic_compile-0.2.0-py3.9.egg/crytic_compile/crytic_compile.py", line 218, in _get_cached_offset_to_line
    self._get_cached_line_to_code(file)
  File "/usr/local/lib/python3.9/site-packages/crytic_compile-0.2.0-py3.9.egg/crytic_compile/crytic_compile.py", line 247, in _get_cached_line_to_code
    source_code = self.src_content[file.absolute]
AttributeError: 'str' object has no attribute 'absolute'
ERROR:root:None
ERROR:root:Error in bsc:0x73feaa1ee314f8c655e354234017be2193c9e24e
ERROR:root:Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/slither/__main__.py", line 723, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
  File "/usr/local/lib/python3.9/site-packages/slither/__main__.py", line 82, in process_all
    ) = process_single(compilation, args, detector_classes, printer_classes)
  File "/usr/local/lib/python3.9/site-packages/slither/__main__.py", line 65, in process_single
    slither = Slither(target, ast_format=ast, **vars(args))
  File "/usr/local/lib/python3.9/site-packages/slither/slither.py", line 81, in __init__
    parser.parse_top_level_from_loaded_json(ast, path)
  File "/usr/local/lib/python3.9/site-packages/slither/solc_parsing/slither_compilation_unit_solc.py", line 180, in parse_top_level_from_loaded_json
    pragma.set_offset(top_level_data["src"], self._compilation_unit)
  File "/usr/local/lib/python3.9/site-packages/slither/core/source_mapping/source_mapping.py", line 124, in set_offset
    self._source_mapping = self._convert_source_mapping(offset, compilation_unit)
  File "/usr/local/lib/python3.9/site-packages/slither/core/source_mapping/source_mapping.py", line 101, in _convert_source_mapping
    (lines, starting_column, ending_column) = self._compute_line(
  File "/usr/local/lib/python3.9/site-packages/slither/core/source_mapping/source_mapping.py", line 40, in _compute_line
    start_line, starting_column = compilation_unit.core.crytic_compile.get_line_from_offset(
  File "/usr/local/lib/python3.9/site-packages/crytic_compile-0.2.0-py3.9.egg/crytic_compile/crytic_compile.py", line 235, in get_line_from_offset
    self._get_cached_offset_to_line(file)
  File "/usr/local/lib/python3.9/site-packages/crytic_compile-0.2.0-py3.9.egg/crytic_compile/crytic_compile.py", line 218, in _get_cached_offset_to_line
    self._get_cached_line_to_code(file)
  File "/usr/local/lib/python3.9/site-packages/crytic_compile-0.2.0-py3.9.egg/crytic_compile/crytic_compile.py", line 247, in _get_cached_line_to_code
    source_code = self.src_content[file.absolute]
AttributeError: 'str' object has no attribute 'absolute'
```

My command is: `slither bsc:0x73feaa1ee314f8c655e354234017be2193c9e24e`

Thanks!

**Comment by a4dyn** [comment]
Update: Even not using but downloading the pancakeswap masterchef source code & compile the same error still occurs. Would you like to help looking into it? Thanks!

**Comment by montyly** [comment]
Hi @a4dyn . Thanks for reporting this. I believe this is now fixed, as `slither bsc:0x73feaa1ee314f8c655e354234017be2193c9e24e` works from master.

The fix will be available with the upcoming release https://github.com/crytic/slither/issues/905

**Comment by 0xalpharush** [comment]
This is included in the latest release 0.8.2. You can install it `pip install slither-analyzer --upgrade`
